src/cpp/src/text_callback_streamer.cpp

```diff
@@ -15,6 +15,14 @@ bool TextCallbackStreamer::put(int64_t token) {
     std::stringstream res;
     m_tokens_cache.push_back(token);
     std::string text = m_tokenizer.decode(m_tokens_cache);
+    
+    // In some cases adding the next token can shorten the text, 
+    // e.g. when apostrophe removing regex had worked after adding new tokens.
+    // Need to hold on before flushing if apostrophe is the last symbol.
+    if (text.size() > 1 && text.back() == '\'') {
+        return on_finalized_subword_callback(res.str());
+    }
```

Comment:
Is it guaranteed that this may happen to `'` only? Maybe the streamer should always keep the last symbol unprinted until `\n` or `end()`

Reply:
We also thought about that. That would've been better than the current solution. 

I am afraid i can make a mistake when i manipulate with `print_len` with the current approach risks are minimal. But let me check with printing only last n - 1 (or n - k) tokens

Reply:
Done. Now the last N characters are not printed just in case if anything will change after the tokens arrive. Set N = 4, even setting 1 already fixes the issue, but to ensure set it a bit bigger. Please take a look.

Reply:
BTW, HF streamer has the same issue with eating characters. @apaniukov can confirm. If we merge this PR we will be one step ahead of HF 🚀